### PR TITLE
chore: Upgrade to Kubernetes 0.30.1 and associated controller-runtime.

### PR DESCRIPTION
**Pull request by hessjcg**
This upgrades the Kuberentes and Controller Runtime libraries to the latest available. It includes fixes for 
superficial breaking changes in the Controller Runtime API.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,16 @@
 module github.com/GoogleCloudPlatform/cloud-sql-proxy-operator
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.3
```
Comment (enocom) [inline go.mod:5]: Do we want this?

File: go.mod [hunk 2]
```diff
@@ -1,14 +1,16 @@
 module github.com/GoogleCloudPlatform/cloud-sql-proxy-operator
 
-go 1.22
+go 1.22.0
```
Comment (enocom) [inline go.mod:3]: Why not just go 1.22?

File: go.mod (hunk 1 above)
Comment (hessjcg) [reply go.mod:5]: It seems this was added automatically by `go mod tidy`. I'll revert it.

File: go.mod (hunk 2 above)
Comment (hessjcg) [reply go.mod:3]: Same here.

File: go.mod (hunk 1 above)
Comment (hessjcg) [reply go.mod:5]: Ok, `go mod tidy` automatically updates:

```
go 1.22
```

to 

```
go 1.22.0
toolchain go1.22.3
```

Apparently, this is the Go official way now. I think this change is necessary.

Comment (enocom) [reply go.mod:5]: This looks like a problem with a dependency we're pulling in. See https://github.com/golang/go/issues/62409 as one example. I think we don't want to add the toolchain or the patch version here.

File: infra/resources/database.tf [hunk 3]
```diff
@@ -42,6 +46,13 @@ resource "google_sql_database" "db" {
   project  = var.project_id
 }
 
+resource "google_sql_user" "iam_user" {
+  name     = "hessjc@google.com"
```
Comment (enocom) [inline infra/resources/database.tf:50]: This looks like a dev artifact?

Comment (hessjcg) [reply infra/resources/database.tf:50]: Good catch. I'll get rid of this.

**Review by enocom — COMMENTED** [review]
My only concern here is the unwanted changes to go.mod. That looks like a bad dependency doing this.

**Comment by hessjcg** [comment]
This is no longer a relevant PR. Newer versions of kubernetes clients were released.

**Pull request closed without merging**